addons/hr_homeworking/models/hr_homeworking.py

```diff
@@ -0,0 +1,137 @@
+# -*- coding: utf-8 -*-
+# Part of Odoo. See LICENSE file for full copyright and licensing details.
+
+from odoo import _, api, fields, models
+from odoo.addons.resource.models.utils import string_to_datetime
+DAYS = ['monday_location_id', 'tuesday_location_id', 'wednesday_location_id', 'thursday_location_id', 'friday_location_id', 'saturday_location_id', 'sunday_location_id']
+
+class HrHomeworking(models.Model):
+    _name = "hr.homeworking"
+
+    def _default_employee(self):
+        if 'default_employee' in self.env.context:
+            return self.env.context.default_employee
+        return self.env['hr.employee'].search([('user_id.id','=', self.env.user.id)])
+        
+    # work location
+    work_location_id = fields.Many2one('hr.work.location', required=True)
+    work_location_name = fields.Char(related='work_location_id.name')
+    work_location_type = fields.Selection(related="work_location_id.icon")
+    #relative to event
+    employee_id = fields.Many2one('hr.employee', default=_default_employee)
+    weekday = fields.Integer()
+    weekly = fields.Boolean(default=False)
+    date = fields.Date(help="useful only if it's a special worklocation", readonly=True)
+    removed = fields.Boolean(default=False, help="True if a specific date on a default record is removed")
+    parent_default_homeworking = fields.Many2one("hr.homeworking", help="removed records from default record")
+    child_removed_ids = fields.One2many("hr.homeworking",'parent_default_homeworking')
+    _sql_constraints = [
+        ('uniq_name', 'unique(employee_id, weekday)', 'Only one default work location per day per employee.'),
+        ('uniq_truc', 'unique(employee_id, weekly, date)', 'Only one default work location and one exceptional work location per day per employee.'),
+    ]
+
+    def action_register_new_day_work_location(self):
+        for record in self:
+            if record.weekly:
+                record.weekday = record.date.weekday()
+                record.employee_id.sudo().write({DAYS[record.weekday]: record.work_location_id})
+
+    def create_exceptional_worklocation(self, vals):
```

Comment:
vals is a confusing argument here, it's like a wildcard, I don't know what will be in it, or what I should pass. Basically, makes this method hard to use for other cases. 

All you are using if for `vals.get("work_location_id")`, so, why won't you just have another argument work_location_id instead